src/PhpSpec/CodeGenerator/Generator/Argument/StringBuilder.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace PhpSpec\CodeGenerator\Generator\Argument;
+
+class StringBuilder
+{
+    /**
+     * @param \ReflectionParameter[] $parameters
+     *
+     * @return string
+     */
+    public function buildFromReflectionParameters(array $parameters, string $targetClassNamespace) : string
+    {
+        return implode(', ', array_map(function (\ReflectionParameter $parameter) use ($targetClassNamespace) {
+            return $this->buildArgument($parameter, $targetClassNamespace);
+        }, $parameters));
+    }
+
+    private function buildArgument(\ReflectionParameter $parameter, string $targetClassNamespace) : string
+    {
+        $parameterName = '$' . $parameter->getName();
+
+        $type = $parameter->getType();
+
+        switch (true) {
+            case (class_exists($type) || interface_exists($type)):
```

Comment:
Why does the class have to exist before we resolve its type? I'd expect this logic to also apply to not existing classes too?

Reply:
The interface method wouldn't be type-hinting for a non-existent class or interface at the stage we're speccing the implementor, so we should check it exists which means we can also use reflection to get its argument list

Reply:
It's very possible to have type hints for types that aren't defined _in the current project yet_

Reply:
i it's completely impossible then you don't need the other case anyhow ;-0